packages/devtools_app/lib/src/config_specific/drag_and_drop/drag_and_drop.dart

```diff
@@ -2,61 +2,169 @@
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
+import 'package:flutter/gestures.dart';
 import 'package:flutter/material.dart';
+import 'package:flutter/rendering.dart';
+import 'package:provider/provider.dart';
 
 import '../../notifications.dart';
+import '../../utils.dart';
 import '_drag_and_drop_stub.dart'
     if (dart.library.html) '_drag_and_drop_web.dart'
     if (dart.library.io) '_drag_and_drop_desktop.dart';
 
-abstract class DragAndDrop extends StatefulWidget {
-  factory DragAndDrop({
-    @required Function(Map<String, dynamic> data) handleDrop,
-    @required Widget child,
-  }) {
-    return createDragAndDrop(
-      handleDrop: handleDrop,
-      child: child,
-    );
+abstract class DragAndDropManager {
+  factory DragAndDropManager({@required NotificationsState notifications}) {
+    final manager = createDragAndDropManager(notifications: notifications);
+    manager.init();
+    return manager;
+  }
+
+  DragAndDropManager.impl({@required this.notifications});
+
+  final NotificationsState notifications;
+
+  final _statesById = <String, DragAndDropState>{};
```

Comment:
I don't think we need this map. Just walking the Element tree and using DragAndDropState is enough. Having the map just leaves rooms for bugs like two widgets with the same id causing a collision in the map and one object to get lost without unregister being called.

Reply:
removed id